barcelona/scraping.py

```diff
@@ -0,0 +1,41 @@
+import urllib.request
+import datetime
+import time
+import json
+
+# Create header of csv file with column names
+# Only has to be ran once
+with open("barcelona/test_scrape.csv", mode="w+") as f:
+    f.write('time;id;type;latitude;longitude;streetName;streetNumber;altitude;slots;bikes;nearbyStations;status')
+    f.write("\n")
+
+# Define function to get json response
+def get_response(url):
+    resp = urllib.request.urlopen(url)
+    data = resp.read()
+    json_dt = json.loads(data)
+    return(json_dt)
+
+
+url_bcn = "http://wservice.viabicing.cat/v2/stations"
+delim = ";"
+
+# a+ for append, as the file is created with the header above
+with open("barcelona/test_scrape.csv", mode="a+") as f:
+
+    # We want to do requests every 5 seconds for a complete day.
+    # 86400 seconds in a day, divided by 5 is 17280
+    for i in range(17280):
+        print(i)
+        # Get json response
+        bicycle_dict = get_response(url_bcn)['stations']
+
```

Comment:
I prefer, in general, `dict.get(key)` over `dict[key]`. The former returns `None` if the key is not present, while the second raises `KeyError`